test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller.go

```diff
@@ -158,31 +175,85 @@ func (r *DockerMachinePoolReconciler) SetupWithManager(ctx context.Context, mgr
 			handler.EnqueueRequestsFromMapFunc(utilexp.MachinePoolToInfrastructureMapFunc(
 				infraexpv1.GroupVersion.WithKind("DockerMachinePool"), ctrl.LoggerFrom(ctx))),
 		).
+		Watches(
+			&infrav1.DockerMachine{},
+			handler.EnqueueRequestsFromMapFunc(dockerMachineToDockerMachinePool),
+		).
 		Watches(
 			&clusterv1.Cluster{},
 			handler.EnqueueRequestsFromMapFunc(clusterToDockerMachinePools),
 			builder.WithPredicates(
 				predicates.ClusterUnpausedAndInfrastructureReady(ctrl.LoggerFrom(ctx)),
 			),
-		).Complete(r)
+		).Build(r)
 	if err != nil {
 		return errors.Wrap(err, "failed setting up with a controller manager")
 	}
+
+	r.recorder = mgr.GetEventRecorderFor("dockermachinepool-controller")
+	r.externalTracker = external.ObjectTracker{
+		Controller: c,
+		Cache:      mgr.GetCache(),
+	}
+
 	return nil
 }
 
-func (r *DockerMachinePoolReconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster, machinePool *expv1.MachinePool, dockerMachinePool *infraexpv1.DockerMachinePool) error {
-	pool, err := docker.NewNodePool(ctx, r.Client, cluster, machinePool, dockerMachinePool)
+func (r *DockerMachinePoolReconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster, machinePool *expv1.MachinePool, dockerMachinePool *infraexpv1.DockerMachinePool) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	dockerMachineList, err := getDockerMachines(ctx, r.Client, *cluster, *machinePool, *dockerMachinePool)
 	if err != nil {
-		return errors.Wrap(err, "failed to build new node pool")
+		return ctrl.Result{}, err
 	}
 
-	if err := pool.Delete(ctx); err != nil {
-		return errors.Wrap(err, "failed to delete all machines in the node pool")
+	if len(dockerMachineList.Items) > 0 {
+		log.Info("DockerMachinePool still has dependent DockerMachines, deleting them first and requeuing", "count", len(dockerMachineList.Items))
+
+		var errs []error
+
+		for _, dockerMachine := range dockerMachineList.Items {
+			if !dockerMachine.GetDeletionTimestamp().IsZero() {
+				// Don't handle deleted child
+				continue
+			}
+
+			if err := r.deleteMachinePoolMachine(ctx, dockerMachine); err != nil {
+				err = errors.Wrapf(err, "error deleting DockerMachinePool %s/%s: failed to delete %s %s", dockerMachinePool.Namespace, dockerMachinePool.Name, dockerMachine.Namespace, dockerMachine.Name)
+				errs = append(errs, err)
+			}
+		}
+
+		if len(errs) > 0 {
+			return ctrl.Result{}, kerrors.NewAggregate(errs)
+		}
+
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
```suggestion
		return ctrl.Result{}, nil
```
We are watching for dockerMachines, no need to requeue after

Reply:
I couldn't get this to work without the requeue since the DockerMachinePool deletion timed out. I think the Watch doesn't get triggered when the resource is completely deleted IIRC, but I'll consult @CecileRobertMichon about this soon.

Reply:
@fabriziopandini this is equivalent to https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/controllers/cluster/cluster_controller.go#L271

let's change `RequeueAfter: 10 * time.Second` to use a const defined in this file just like RequeueAfter: deleteRequeueAfter

Reply:
If this is the only thing blocking this PR, you can create an issue to remove the requeue after and assign me, I can take a closer look in the coming week to see if it's possible to remove the requeue since you're going to be out starting tomorrow @Jont828